### DISCUSSION TITLE
error dwv

### DISCUSSION THREAD

**Discussion by manuelatito** [teamalgoritma/community#857] [Q&A] [answered]
Slmt malam, bagaimana mengatasi error di bawah ini? Terima kasih 

![image](https://github.com/teamalgoritma/community/assets/164650405/9ded0a76-d4e6-46f7-a6cb-6c1bf5cfb39c)

![image](https://github.com/teamalgoritma/community/assets/164650405/2155564f-884b-4449-a434-fdb2c1f7b92b)

**Comment by ramdhinafinita** [comment]
Dear @manuelatito 
Terimakasih atas pertanyaanya, jika dilihat dari error yang muncul dikarenakan pada kolom InvoiceMonth bukan dalam index sepertinya input kode yang dilampirkan tidak sesuai dengan eror yang dilampirkan. Namun memang berimpact pada kode terakhir.
Bisa cek kembali terlebih dahulu output dataframe invoice_int nya apakah sudah sesuai dan mungkin bisa dilampirkan beserta output df nya. 
Kurang lebih output seharusnya seperti ini untuk dataframe invoice_int nya:
![image](https://github.com/teamalgoritma/community/assets/64061969/291ca4c6-736f-4f0a-a37e-ebedc2991b5c)


Terimakasih

**Reply by manuelatito** [reply]
Dear kak @ramdhinafinita,
Terima kasih kak atas jawabannya, sy sudah coba kak, ternyata tadi memang ada salah karena belum disimpan dgn variabel invoice_int. Tp setelah saya save dgn variabel invoice_int tetap masih error kak untuk yg no 6 nya. 

![image](https://github.com/teamalgoritma/community/assets/164650405/034aaecb-96c3-4234-b934-800601f14e36)

![image](https://github.com/teamalgoritma/community/assets/164650405/e4f2aace-37f5-4328-9f83-05b44971d8d7)

![image](https://github.com/teamalgoritma/community/assets/164650405/381db1df-4c50-46ea-968c-758e4b6fd141)

Terima kasih

**Reply by ramdhinafinita** [reply]
Apakah sudah cek tipe data pada kolom Invoicedate nya sudah datetime64 bu? 
sepertinya kalau sudah berhasil membuat kolom baru Quarter sudah benar bu, coba bisa cek bagian dataframe june_outliers nya?

**Reply by manuelatito** [reply]
sudah datetime64 kak

![image](https://github.com/teamalgoritma/community/assets/164650405/a4287a2a-8f3a-490c-bbc4-cd0933ae6f6e)

**Comment by ramdhinafinita** [comment]
Bisa di cek terlebih dahulu apakah dataframe june_outliers nya sudah sesuai? 
Karena sepertinya error muncul bukan dari daraframe int.

mungkin cara cek nya seperti ini dulu bu, untuk melihat apakah error muncul dari .loc:
```
invoice_int['Quarter'] = invoice_int['InvoiceDate'].dt.to_period('Q')
invoice_topq = invoice_int.groupby(['Country','Quarter']).\
                 agg({'TotalPrice' : 'sum'})#.loc[june_outliers.axes[0],]
invoice_topq
```

**Reply by manuelatito** [reply]
Kak, dari code kk sudah saya coba dan berhasil, tapi kenapa yg code dari kk utk soal no.6 itu tetap error ya kak?

![image](https://github.com/teamalgoritma/community/assets/164650405/8f9c580d-ff33-462a-afeb-f61b187c900f)

![image](https://github.com/teamalgoritma/community/assets/164650405/59fe4de9-bbec-4bad-ae2c-115e81e7887a)

Terima kasih

**Reply by manuelatito** [reply]
Kak, untuk dataframe di june_outliers sy seperti ini kak

![image](https://github.com/teamalgoritma/community/assets/164650405/26dfcf9e-63d2-4795-8468-915021b25956)

Terima kasih

**Reply by ramdhinafinita** [reply, marked as answer]
Baik, ternyata memang terdapat kesalahan pada dataframe june_outliers bu. Seharusnya output index baris nya adalah nama-nama negara yang termasuk kedalam outliers. Terdapat kesalahan kode untuk menentukan paramater key nya. bisa di cek kembali ya bu. Karena pada soal dijelaskan bahwa mencari dengan kata kunci 'June 2011'

**Reply by manuelatito** [reply]
Baik kak, terima kasih kak Fini atas arahannya, akan sy coba..
